pandas/core/dtypes/generic.py

```diff
@@ -57,6 +57,8 @@ def _check(cls, inst):
 ABCDateOffset = create_pandas_abc_type("ABCDateOffset", "_typ",
                                        ("dateoffset",))
 ABCInterval = create_pandas_abc_type("ABCInterval", "_typ", ("interval", ))
+ABCExtensionArray = create_pandas_abc_type("ABCExtensionArray", "_typ",
+                                           ("extension",))
```

Comment:
So in practice, this `ABCExtensionArray` will only work for externally defined extension arrays? (as for the internal ones they have already another `_type`, like `Categorical`) 
I am not really sure that is actually what we want? As then you can't use this to check if an array-like is an instance of ExtensionArray.

Reply:
or, you can add here 'categorical' to the list of possible `_typ`'s ?

Reply:
Good catch. We'll need to continue to update that as we make our other internal extension arrays.